metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -588,6 +589,11 @@ class MetalsLanguageServer(
     // Update in-memory buffer contents from LSP client
     buffers.put(path, params.getTextDocument.getText)
     trees.didChange(path)
+
+    if (createdFiles.remove(path)) {
```

Comment:
I don't find this particularly readable, I think it would be clearer to split this into two statements.

Reply:
That's aa atomic method on the ConcurrentSet, so I think we would be safer this way. Might be overdoing it though.

Reply:
Yep, I meant exactly what @marek1840 said